### PR TITLE
Group invocations according to grouping strategy

### PR DESCRIPTION
**Pull request by sultan-duisenbay**
Added the functionality to group the results based on the group ID in #1242.
Added unit tests for the functionality above.
Renamed some variables for better clarity

### CODE REVIEW

File: pkg/updater/resultstore/resultstore.go [hunk 1]
```diff
@@ -103,38 +105,76 @@ func ResultStoreColumnReader(client *DownloadClient, reprocess time.Duration) up
...
+			groupedInvocations[groupIdentifier] = group
+		}
+		group.Invocations = append(group.Invocations, invocation)
+	}
+
+	for _, group := range groupedInvocations {
+		sortedGroups = append(sortedGroups, group)
+	}
+
+	// reverse sort groups by invocation time
```
Comment (michelle192837) [inline pkg/updater/resultstore/resultstore.go:153]: nit: I think sorting in the function is a bit opaque (especially if the function doc doesn't clarify it's sorting). imo sorting should happen more local to where it's needed, or this function should clarify it sorts the invocations.

File: pkg/updater/resultstore/resultstore_test.go [hunk 2]
```diff
@@ -1359,3 +1359,504 @@ func TestIncludeStatus(t *testing.T) {
 		})
 	}
 }
+
+func TestGroupInvocations(t *testing.T) {
+	cases := []struct {
+		name        string
+		tg          *configpb.TestGroup
+		invocations []*invocation
+		want        []*invocationGroup
```
Comment (michelle192837) [inline pkg/updater/resultstore/resultstore_test.go:1368]: nit: Does it make sense to make the 'want' argument more compact? I think all this cares about is that each group ID has the right invoc IDs, and I think it'll be more readable.

**Review by michelle192837 — APPROVED** [review]
Mostly LGTM aside from some comments. Thanks!

/hold

Unhold at will

File: pkg/updater/resultstore/resultstore.go (hunk 1 above)
Comment (sultan-duisenbay) [reply pkg/updater/resultstore/resultstore.go:154]: Added a comment to the function

File: pkg/updater/resultstore/resultstore_test.go (hunk 2 above)
Comment (sultan-duisenbay) [reply pkg/updater/resultstore/resultstore_test.go:1368]: Good suggestion - I made the tests more compact by removing some parts which didn't matter

**Pull request merged**